stripe/src/main/java/com/stripe/android/paymentsheet/model/StripeIntentValidator.kt

```diff
@@ -0,0 +1,52 @@
+package com.stripe.android.paymentsheet.model
+
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+
+/**
+ * Validator for [PaymentIntent] or [SetupIntent] instances used in PaymentSheet.
+ */
+internal class StripeIntentValidator {
+    @JvmSynthetic
+    fun requireValid(
+        stripeIntent: StripeIntent
+    ): StripeIntent {
+        when {
+            stripeIntent is PaymentIntent &&
+                stripeIntent.confirmationMethod != PaymentIntent.ConfirmationMethod.Automatic -> {
+                error(
+                    """
+                        PaymentIntent with confirmation_method='automatic' is required.
+                        See https://stripe.com/docs/api/payment_intents/object#payment_intent_object-confirmation_method
+                    """.trimIndent()
+                )
+            }
+            stripeIntent is PaymentIntent &&
+                stripeIntent.status != StripeIntent.Status.RequiresPaymentMethod -> {
+                error(
+                    """
+                        A PaymentIntent with status='requires_payment_method' is required.
+                        The current PaymentIntent has status ${stripeIntent.status}.
+                        See https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status
+                    """.trimIndent()
+                )
+            }
+            stripeIntent is SetupIntent &&
```

Comment:
@mshafrir Is this all we should validate?

Reply:
@mshafrir-stripe